File: tests/basic/matrix_delete.ml
open Optitrust
open Target

let _ = Flags.check_validity := true

let _ = Run.doc_script_cpp (fun _ ->
  !! Matrix_basic.delete ~var:"a" [cFunBody "main"];
)

"
#include \"../../include/optitrust.h\"

int main() {
  int* const a = (int* const) MALLOC1(10, sizeof(int));
  MFREE1(10, a);
  return 0;
}
"

let _ = Run.script_cpp (fun _ ->
  !! Matrix_basic.delete ~var:"a" [cFunBody "main"];
  !! Matrix_basic.delete ~var:"b" [cFunBody "main"];
)

File: tests/basic/matrix_delete.cpp
#include <stdio.h>
#include "../../include/optitrust.h"

extern int N;
typedef float T;

int main() {
  T* a = (T*) MALLOC1(N, sizeof(T));
  T* b = (T*) MALLOC1(N - 2, sizeof(T));
  for (int i = 0; i < N; i++) {
    printf("%i\n", i);
  }

  MFREE1(N, a);
  MFREE1(N - 2, b);
  return 0;
}